File: 1028/1028.java
// bee 1028 - Figurinhas
// Alonso Martins
// 05/04/2023
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        
        for (int i = 0; i < n; i++) {
            int f1 = sc.nextInt();
            int f2 = sc.nextInt();
            int gcd = gcd(f1, f2); //calcula o MDC
            
            System.out.println(gcd);
        }
        
        sc.close();
    }
    
    //Método para calcular o MDC (Máximo Divisor Comum) entre dois números
    public static int gcd(int a, int b) {
        if (b == 0) {
            return a;
        }
        
        return gcd(b, a % b);
    }
}

File: 1028/1028.c
// bee 1028 - Figurinhas
// Alonso Martins
// 09/04/2024
#include <stdio.h>

int mdc(int a, int b)
{
	int temp;
	while (b != 0)
	{
		temp = b;
		b = a % b;
		a = temp;
	}
	return a;
}
int main()
{
    int N, f1, f2, i;

    scanf("%d", &N);

    while (N --)
    {
        scanf("%d %d", &f1, &f2);

        printf("%d\n", mdc(f1, f2));
    }

    return 0;
}
